models/hh_cond_beta_gap_traub.cpp

```diff
@@ -0,0 +1,787 @@
+  assert( from < to );
+
+  const size_t interpolation_order =
+    kernel().simulation_manager.get_wfr_interpolation_order();
+  const double wfr_tol = kernel().simulation_manager.get_wfr_tol();
+  bool wfr_tol_exceeded = false;
+
+  // allocate memory to store the new interpolation coefficients
+  // to be sent by gap event
+  const size_t buffer_size =
+    kernel().connection_manager.get_min_delay() * ( interpolation_order + 1 );
+  std::vector< double > new_coefficients( buffer_size, 0.0 );
+
+  // parameters needed for piecewise interpolation
+  double y_i = 0.0, y_ip1 = 0.0, hf_i = 0.0, hf_ip1 = 0.0;
+  double f_temp[ State_::STATE_VEC_SIZE ];
+
+  for ( long lag = from; lag < to; ++lag )
+  {
+
+    // B_.lag is needed by hh_cond_beta_gap_traub_dynamics to
+    // determine the current section
+    B_.lag_ = lag;
+
+    if ( called_from_wfr_update )
+    {
+      y_i = S_.y_[ State_::V_M ];
+      if ( interpolation_order == 3 )
+      {
+        hh_cond_beta_gap_traub_dynamics(
+          0, S_.y_, f_temp, reinterpret_cast< void* >( this ) );
+        hf_i = B_.step_ * f_temp[ State_::V_M ];
+      }
+    }
+
+    double t = 0.0;
+    const double U_old = S_.y_[ State_::V_M ];
+
+    // numerical integration with adaptive step size control:
+    // ------------------------------------------------------
+    // gsl_odeiv_evolve_apply performs only a single numerical
+    // integration step, starting from t and bounded by step;
+    // the while-loop ensures integration over the whole simulation
+    // step (0, step] if more than one integration step is needed due
+    // to a small integration step size;
+    // note that (t+IntegrationStep > step) leads to integration over
+    // (t, step] and afterwards setting t to step, but it does not
+    // enforce setting IntegrationStep to step-t; this is of advantage
+    // for a consistent and efficient integration across subsequent
+    // simulation intervals
+    while ( t < B_.step_ )
+    {
+      const int status = gsl_odeiv_evolve_apply( B_.e_,
+        B_.c_,
+        B_.s_,
+        &B_.sys_,             // system of ODE
+        &t,                   // from t
+        B_.step_,             // to t <= step
+        &B_.IntegrationStep_, // integration step size
+        S_.y_ );              // neuronal state
+      if ( status != GSL_SUCCESS )
+      {
+        throw GSLSolverFailure( get_name(), status );
+      }
+    }
+
+    if ( not called_from_wfr_update )
+    {
+      S_.y_[ State_::DG_EXC ] +=
+        B_.spike_exc_.get_value( lag ) * V_.PSConInit_E;
+      S_.y_[ State_::DG_INH ] +=
+        B_.spike_inh_.get_value( lag ) * V_.PSConInit_I;
+      // sending spikes: crossing 0 mV, pseudo-refractoriness and local
+      // maximum...
+      // refractory?
+      if ( S_.r_ > 0 )
+      {
+        --S_.r_;
+      }
+      else
+        // (    threshold    &&     maximum       )
+        if ( S_.y_[ State_::V_M ] >= P_.V_T + 30.
```

Comment:
Where does the value 30 come from? Can we give this a name and move it to a private const member variable?

Reply:
It is exactly the same code as in `hh_cond_exp_traub`, so I suggest to leave this as it its.